src/components/EmailForm/index.tsx

```diff
@@ -37,79 +36,56 @@ export const EmailForm: React.FC<EmailFormProps> = ({ eventName, mail }) => {
+    try {
+      const response = await postMessage(messageDataFormatted);
 
-    const mailLink = document.createElement('a');
-    mailLink.href = `mailto:${mail}?subject=${encodedSubject}&body=${encodedBody}`;
-    mailLink.click();
+      if (response && !response.error) {
+        setStatus(200);
+      }
 
-    setTimeout(() => {
-      if (window?.document.hasFocus()) {
-        setIsModalOpen(!isModalOpen);
+      if (response?.error) {
+        setStatus(response.error?.status);
       }
-    }, 1000);
+    } finally {
+      setIsLoading(false);
+      setIsModalOpen(true);
+    }
+  };
+
+  const handleClose = () => {
+    setIsModalOpen(false);
   };
 
   return (
-    <FormProvider {...formProps}>
-      <form
-        onSubmit={handleSubmit(onSubmit)}
-        className={cx('mt-6 flex flex-col', {
-          hidden: isModalOpen,
-        })}
-      >
-        <Textarea
-          register={register}
-          type="text"
-          name="email_body"
-          label="Emailová zpráva"
-          placeholder="Ahoj, můžeš mi poslat bližší informace..."
-          errors={errors}
-        />
-        <button
-          type="submit"
-          className="ml-auto mt-5 h-11 w-40 rounded-md bg-button py-2 text-white hover:text-primary focus:text-primary"
-        >
-          Zúčastnit se
-        </button>
-      </form>
-      <article
-        className={cx(
-          'relative mt-10 flex flex-col items-center justify-center bg-card p-7 shadow-md',
-          {
-            hidden: !isModalOpen,
-          }
-        )}
-      >
-        <h3 className="mb-6 max-w-md text-center text-lg text-secondary">
-          Vypadá to, že váš emailový klient není správně nastaven. Prosím,
-          zadejte údaje ručně.
-        </h3>
-        <table className="text-s mb-6 w-full font-medium">
-          <tbody>
-            <tr className="border-b border-low-contrast">
-              <td className="py-2 md:py-4">Email</td>
-              <td className="py-2 font-light md:py-4">{emailData.mail}</td>
-            </tr>
-            <tr className="border-b border-low-contrast">
-              <td className="py-2 md:py-4">Předmět emailu:</td>
-              <td className="py-2 font-light md:py-4">{emailData.subject}</td>
-            </tr>
-            <tr className="border-b border-low-contrast">
-              <td className="py-2 md:py-4">Obsah emailu:</td>
-              <td className="max-w-md py-2 font-light md:py-4">
-                {emailData.body.email_body}
-              </td>
-            </tr>
-          </tbody>
-        </table>
-      </article>
-    </FormProvider>
+    <>
+      <FormProvider {...formProps}>
+        <form onSubmit={handleSubmit(onSubmit)} className="mt-6 flex flex-col">
+          <Textarea
+            register={register}
+            type="text"
+            name="text"
+            label="Emailová zpráva"
+            placeholder="Ahoj, můžeš mi poslat bližší informace..."
+            errors={errors}
```

Comment:
Is there any chance errors could be returned from `useFormContext` or `useWatch` inside so we wouldn't have to pass it inside every input?

Reply:
Done.